src/test/resources/refaster/ArraysRecipe.java

```diff
@@ -53,8 +60,12 @@ public String getDescription() {
     @Override
     public TreeVisitor<?, ExecutionContext> getVisitor() {
         JavaVisitor<ExecutionContext> javaVisitor = new AbstractRefasterJavaVisitor() {
-            final JavaTemplate before = Semantics.expression(this, "before", (String[] strings) -> String.join(", ", strings)).build();
-            final JavaTemplate after = Semantics.expression(this, "after", (String[] strings) -> String.join(":", strings)).build();
+            final JavaTemplate before = JavaTemplate
+                    .builder("String.join(\", \", #{strings:any(java.lang.String[])})")
```

Comment:
I see, we have an explicit exception for `java.lang`: https://github.com/openrewrite/rewrite-templating/blob/261ad17a7393fa5db5553b8a58e3a646343ee4ea/src/main/java/org/openrewrite/java/template/internal/TemplateCode.java#L118

It does make the template code a bit more readable, but it could cause problems when the Java class an after template gets embedded into is in a package which has a class by the same name, which is unusual but not unheard of.

I am a bit on the fence on this one. `java.lang` is a special case also in the `ShortenFullyQualifiedTypeReferences` recipe (for basically the same reason). But I think in probably around 90% of the cases the name would end up getting shortened, because the compilation unit already has another (unqualified) reference to the same type in which case the recipe knows it can shorten the reference.

As I pointed out before, this isn't directly related to your PR. It is just something that caught my eye.

Reply:
Yes we can tackle this separately; there's a few other items too I'd been looking to do after this lands, since any changes previously were likely to conflict with this one.